warrior/Actions/CiRegressionActions/ci_regression_actions.py

```diff
@@ -329,3 +329,99 @@ def delete_tmp_dir(self):
+            2. int_value - expected to be int
+            3. float_value - expected to be float
+            4. bool_value - expected to be bool
+            5. list_value - expected to be list
+            6. tuple_value - expected to be tuple
+            7. dict_value - expected to be dict
+            8. file_value - expected to be file
+        """
+        status = True
+        err_msg = "{} is not an {} value but of type {}"
+        if type(str_value) is not str:
+            print_error(err_msg.format(str_value, "str", type(str_value)))
+            status = False
+        if type(int_value) is not int:
+            print_error(err_msg.format(int_value, "int", type(int_value)))
+            status = False
+        if type(float_value) is not float:
+            print_error(err_msg.format(float_value, "float", type(float_value)))
+            status = False
+        if type(bool_value) is not bool:
+            print_error(err_msg.format(bool_value, "bool", type(bool_value)))
+            status = False
+        if type(list_value) is not list:
+            print_error(err_msg.format(list_value, "list", type(list_value)))
+            status = False
+        if type(tuple_value) is not tuple:
+            print_error(err_msg.format(tuple_value, "tuple", type(tuple_value)))
+            status = False
+        if type(dict_value) is not dict:
+            print_error(err_msg.format(dict_value, "dict", type(dict_value)))
+            status = False
+        if type(file_value) is not file:
+            print_error(err_msg.format(file_value, "file", type(file_value)))
+            status = False
+        return status
+
+    def check_values_from_datafile(self, system_name, strvar, langs, states,
+                                   currencys, ramspace, configfile, intvar,
+                                   file_config):
+        """get the values from datafile
+        :Argument:
+            1. system_name = system name in the datafile
+            2. strvar = string variable
+            3. langs = list variable (should get from data file using wtag)
+            4. states = tuple variable
+            5. currencys = dict variable
+            6. ramspace = boolean variable
+            7. configfile = file variable
+            8. intvar = int variable
+            9. file_config = file variable
+        """
+        def check_type(var, varname, datatype):
+            """check that vars are of correct datatype
+            """
+            vartype = type(var)
+            status = True
+            if vartype is not datatype:
+                print_error('{} is expected to be {} type, but found to be of '
+                            '{} type'.format(varname, datatype, vartype))
+                status = False
+            return status
+        status = True
+        datafile = Utils.config_Utils.datafile
+        tc_filepath = os.path.dirname(data_Utils.get_object_from_datarepository(
```

Comment:
Please add inline comments to indicate which part of the code is doing the check for a particular type of value
for example
`#this block checks if value is string type
status = check_type(strvar, "strvar", str) and status
#this block checks if value is list type
status = check_type(langs, "langs", list) and status
`

Reply:
done.  Please check and approve.

Reply:
Hi @navvrnd 
What about L431-448
I am still not sure what that part of the code is trying to test

Reply:
Hi @kaheichan that is to test the existing tag functionality which should not be broken because of the new wtag functionality

Reply:
I see, understood.
A suggestion for the valid file check, let's add another line of os.path.isfile to enhance the check